plenum/test/primary_selection/test_catchup_multiple_rounds.py

```diff
@@ -0,0 +1,64 @@
+from plenum.common.constants import DOMAIN_LEDGER_ID
+from plenum.test.delayers import delay_3pc_messages, icDelay
+from plenum.test.helper import sendReqsToNodesAndVerifySuffReplies, \
+    sendRandomRequests, waitForSufficientRepliesForRequests, checkViewNoForNodes
+from plenum.test.node_catchup.helper import waitNodeDataEquality
+from plenum.test.pool_transactions.conftest import clientAndWallet1, \
+    client1, wallet1, client1Connected, looper, nodeThetaAdded, \
+    stewardAndWallet1, steward1, stewardWallet
+from plenum.test.batching_3pc.conftest import tconf
+
+from plenum.test.test_node import getNonPrimaryReplicas, getPrimaryReplica, \
+    checkProtocolInstanceSetup
+from plenum.test.view_change.helper import ensure_view_change
+from stp_core.loop.eventually import eventually
+
+Max3PCBatchSize = 3
+TestRunningTimeLimitSec = 150
+
+
+def test_slow_nodes_catchup_before_selecting_primary_in_new_view(tconf,
+                                                                 looper,
+                                                                 txnPoolNodeSet,
+                                                                 client1,
+                                                                 wallet1,
+                                                                 one_node_added,
+                                                                 client1Connected):
+    """
+    Delay 3PC messages to one node and view change messages to some others
+    (including primary)so the node that does not receive enough 3PC messages is
+    behind but learns of the view change quickly and starts catchup,
+    """
+    new_node = one_node_added
+    nprs = [r.node for r in getNonPrimaryReplicas(txnPoolNodeSet, 0)]
+    primary_node = getPrimaryReplica(txnPoolNodeSet, 0).node
+    slow_node = nprs[-1]
+    # nodes_slow_to_inst_chg = [primary_node] + nprs[:2]
+    nodes_slow_to_inst_chg = [n for n in txnPoolNodeSet if n != slow_node]
+    delay_3pc = 10
+    delay_ic = 5
+    delay_3pc_messages([slow_node], 0, delay_3pc)
+    for n in nodes_slow_to_inst_chg:
+        n.nodeIbStasher.delay(icDelay(delay_ic))
+
+    start_count = lambda: sum([1 for e in slow_node.ledgerManager.spylog.getAll(
+                    slow_node.ledgerManager.startCatchUpProcess.__name__)
+                    if e.params['ledgerId'] == DOMAIN_LEDGER_ID])
+
+    s = start_count()
+    requests = sendRandomRequests(wallet1, client1, 10*Max3PCBatchSize)
+
+    ensure_view_change(looper, nodes=txnPoolNodeSet,
+                       exclude_from_check=nodes_slow_to_inst_chg)
+
+    waitForSufficientRepliesForRequests(looper, client1,
+                                        requests=requests)
+    e = start_count()
+    assert e - s >= 2
```

Comment:
Why the slow node is expected to perform more than two catch-ups?
One catch-up because of view change, but what will cause the other catch-up? Or is it expected that the slow nodes needs two rounds of catch-up during view change?